modules/brainbrowser/js/brainbrowser.loris.js

```diff
@@ -836,7 +837,7 @@ $(function() {
     /////////////////////
     // Load the volumes.
     /////////////////////
-    request.then(function(filename) {
+    $.when.apply($, requests).then(function() {
```

Comment:
What exactly does `$.when.apply` do? Is it waiting for all the `requests` to resolve and then triggering the function? 

Eventually I think it would be preferable to switch to using JavaScript promises to handle this sort of thing, but I think that would involve major refactoring of this file.

Reply:
@HenriRabalais [$.when.apply](https://stackoverflow.com/questions/14777031/what-does-when-apply-somearray-do) you're correct. Maybe update to forEach with promises can be done after release?

Reply:
Why would it require major refactoring? It seems to me that this is just 1 (new) request chain which should be stand alone relative to the rest of the JS in the file.

Reply:
@driusan Sorry, I meant the switch from jquery to javascript would probably take a lot of refactoring. This change could be done (relatively?) easily I think. Just wondering if it's better to keep it with jQuery if that's what the rest of the file is already using.

Reply:
My personal thoughts are that it's worth using a real promise chain here since it's part of the javascript standard and doesn't interfere with anything else, rather than getting more-and-more dependent on jQuery.